src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -727,7 +745,23 @@ export class RecastJSCrowd implements ICrowd {
 
         // update transforms
         for (let index = 0; index < this.agents.length; index++) {
-            this.transforms[index].position = this.getAgentPosition(this.agents[index]);
+            // update transform position
+            const agentIndex = this.agents[index];
+            const agentPosition = this.getAgentPosition(agentIndex);
+            this.transforms[index].position = agentPosition;
+            // check agent reach destination
+            if (this.agentDestinationArmed[index]) {
+                const dx = agentPosition.x - this.agentDestination[index].x;
+                const dz = agentPosition.z - this.agentDestination[index].z;
+                const radius = this.reachRadii[index];
+                const groundY = this.agentDestination[index].y - this.reachRadii[index];
+                const ceilingY = this.agentDestination[index].y + this.reachRadii[index];
+                const distanceXZSquared = dx * dx + dz * dz;
+                if (agentPosition.y > groundY && agentPosition.y < ceilingY && distanceXZSquared < (radius * radius)) {

```

Comment:
Why making a specific case for the y coordinate and not simply checking that `distance_squared(agent, agentDestination) < radius²`?

Reply:
Agents are cylinders.